packages/react/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -83,6 +104,7 @@ const ButtonGroup = React.forwardRef<HTMLElement, ButtonGroupProps>(function But
   forwardRef,
 ) {
   const enabled = useFeatureFlag('primer_react_css_modules_ga')
+  const buttons = React.Children.map(children, (child, index) => <div key={index}>{child}</div>)
```

Comment:
One alternative idea I had here was to check when the child is the last item, we could add a special classname to re-round the corners, thinking it might be more predictable than relying on direct children selectors. But this also works and I'm fine with it getting out like this.